File: BOJ/1026.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;
int n,sum;
int a[50];
int b[50];
int main(){
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {scanf("%d", &a[i]);}
	for (int i = 0; i < n; i++) { scanf("%d", &b[i]);}
	sort(a, a + n);
	sort(b, b + n);
	for (int i = 0, j = n - 1; i < n; i++, j--) {
		sum += (a[j] * b[i]);
	}
	printf("%d\n", sum);
	return 0;
}

File: BOJ(Python 공부용)/1026.py
#!/usr/bin/env python

import sys

sys.setrecursionlimit(50000000)

n = int(sys.stdin.readline())
a = list(map(int, sys.stdin.readline().split(' ')))
b = list(map(int, sys.stdin.readline().split(' ')))

a.sort()
b.sort()
sum = 0
j = len(b) - 1
for i in range(0, len(a)):
    sum += a[i] * b[j]
    j -= 1

print(sum)
